client/my-sites/site-settings/manage-connection/disconnect-site.jsx

```diff
@@ -0,0 +1,154 @@
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import DisconnectJetpackDialog from 'blocks/disconnect-jetpack-dialog';
+import QuerySitePlans from 'components/data/query-site-plans';
+import SiteToolsLink from 'my-sites/site-settings/site-tools/link';
+import { recordGoogleEvent } from 'state/analytics/actions';
+import { disconnect } from 'state/jetpack/connection/actions';
+import { disconnectedSite as disconnectedSiteDeprecated } from 'lib/sites-list/actions';
+import { setAllSitesSelected } from 'state/ui/actions';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { getPlanClass } from 'lib/plans/constants';
+import { successNotice, errorNotice, infoNotice, removeNotice } from 'state/notices/actions';
+
+class DisconnectSite extends Component {
+	state = {
+		dialogVisible: false,
+	}
+
+	handleClick = ( event ) => {
+		event.preventDefault();
+
+		this.setState( {
+			dialogVisible: true,
+		} );
+	}
+
+	handleHideDialog = () => {
+		this.setState( {
+			dialogVisible: false
+		} );
+	}
+
+	disconnectJetpack = () => {
+		const {
+			site,
+			siteId,
+			translate,
+			successNotice: showSuccessNotice,
+			errorNotice: showErrorNotice,
+			infoNotice: showInfoNotice,
+			removeNotice: removeInfoNotice,
+			disconnect: disconnectSite,
+			recordGoogleEvent: recordGAEvent
+		} = this.props;
+
+		this.setState( {
+			dialogVisible: false
+		} );
+
+		recordGAEvent( 'Jetpack', 'Clicked To Confirm Disconnect Jetpack Dialog' );
+
+		const { notice } = showInfoNotice(
+			translate( 'Disconnecting %(siteName)s.', { args: { siteName: site.title } } ),	{
+				isPersistent: true,
+				showDismiss: false,
+			}
+		);
+
+		disconnectSite( siteId ).then( () => {
+			// Removing the domain from a domain-only site results
+			// in the site being deleted entirely. We need to call
+			// `receiveDeletedSiteDeprecated` here because the site
+			// exists in `sites-list` as well as the global store.
+			disconnectedSiteDeprecated( site );
+			this.props.setAllSitesSelected();
+			removeInfoNotice( notice.noticeId );
+			showSuccessNotice( translate( 'Successfully disconnected %(siteName)s.', { args: { siteName: site.title } } ) );
+			recordGAEvent( 'Jetpack', 'Successfully Disconnected' );
+		}, () => {
+			removeInfoNotice( notice.noticeId );
+			showErrorNotice( translate( '%(siteName)s failed to disconnect', { args: { siteName: site.title } } ) );
+			recordGAEvent( 'Jetpack', 'Failed Disconnected Site' );
+		}, );
+
+		page.redirect( '/stats' );
+	}
+
+	render() {
+		const {
+			planClass,
+			site,
+			siteId,
+			siteSlug,
+			translate
+		} = this.props;
+
+		if ( ! site ) {
+			return null;
+		}
+
+		return (
+			<div>
+				<QuerySitePlans siteId={ siteId } />
+
+				<SiteToolsLink
+					href="#"
+					onClick={ this.handleClick }
+					title={ translate( 'Disconnect from WordPress.com' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'Disconnect from WordPress.com?' )` *ES Score: 17*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).